src/App.tsx

```diff
@@ -1,14 +1,60 @@
-import React from 'react';
+import React, {
+  useState,
+  useCallback,
+  useMemo,
+} from 'react';
 import './App.scss';
+import debounce from 'lodash.debounce';
 import { peopleFromServer } from './data/people';
+import { List } from './components/PeopleDropDownList';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const copyPeople = [...peopleFromServer];
+  const [query, setQuery] = useState('');
+  const [focused, setFocused] = useState(false);
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [delayedQuery, setQuerySearch] = useState('');
+  const [trigger, setTrigger] = useState('');
+
+  const filteredPeople = useMemo(() => {
+    return copyPeople.filter(
+      person => person.name.toLowerCase().includes(
+        delayedQuery.toLowerCase(),
+      ),
+    );
+  }, [copyPeople, query]);
+  const aplydelayedQuery = useCallback(
+    debounce(setQuerySearch, 1000), [],
+  );
+
+  const queryWatcher = useCallback(
+    debounce(setTrigger, 1000), [query],
+  );
+  const inputHandler = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    aplydelayedQuery(event.target.value);
+    queryWatcher(event.target.value);
+  };
+
+  const onClick = (person: Person) => {
+    debounce(() => {
+      setQuery(person.name);
+    }, 1000)();
+    debounce(() => {
+      setSelectedPerson(person);
+    }, 1000)();
```

Comment:
why do you need to use double debounce here?
```suggestion
    debounce(() => {
      setQuery(person.name);
      setSelectedPerson(person);
    }, 1000)();
```